src/Datadog.Trace/ClrProfiler/AutoInstrumentation/ILogger/DatadogLoggingScope.cs

```diff
@@ -0,0 +1,97 @@
+﻿// <copyright file="DatadogLoggingScope.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.ILogger
+{
+    internal class DatadogLoggingScope : IReadOnlyList<KeyValuePair<string, object>>
+    {
+        private readonly string _service;
+        private readonly string _env;
+        private readonly string _version;
+        private readonly Tracer _tracer;
+
+        public DatadogLoggingScope()
+            : this(Tracer.Instance)
+        {
+        }
+
+        internal DatadogLoggingScope(Tracer tracer)
+        {
+            _tracer = tracer;
+            _service = tracer.DefaultServiceName ?? string.Empty;
+            _env = tracer.Settings.Environment ?? string.Empty;
+            _version = tracer.Settings.ServiceVersion ?? string.Empty;
+        }
+
+        public int Count => 5;
+
+        public KeyValuePair<string, object> this[int index]
+        {
+            get
+            {
+                if (index == 0)
+                {
+                    return new KeyValuePair<string, object>("dd_service", _tracer.ActiveScope?.Span.ServiceName ?? _service);
+                }
+                else if (index == 1)
+                {
+                    return new KeyValuePair<string, object>("dd_env", _env);
+                }
+                else if (index == 2)
+                {
+                    return new KeyValuePair<string, object>("dd_version", _version);
+                }
+                else if (index == 3)
+                {
+                    return new KeyValuePair<string, object>("dd_trace_id", _tracer.ActiveScope?.Span.TraceId);
+                }
+                else if (index == 4)
+                {
+                    return new KeyValuePair<string, object>("dd_span_id", _tracer.ActiveScope?.Span.SpanId);
+                }
+
+                throw new ArgumentOutOfRangeException(nameof(index));
+            }
+        }
```

Comment:
NIT: should probably be a switch case. The compiler has special optimizations for consecutive numerical cases

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/1663/commits/039f1a3f0085e6215bed7ecb1460019bf7f9e1ac